test/functional/testlib.bash

```diff
@@ -1366,7 +1366,15 @@ destroy_test_environment() { # swupd_function
 			EOM
 		return
 	fi
-	validate_path "$env_name"
+	validate_param "$env_name"
+
+	# if the test environment doesn't exist warn the user but don't terminate script
+	# execution, this function could be called from a test teardown and terminating
+	# at that point will cause incorrect test reporting
+	if [ ! -d "$env_name" ]; then
+		echo "Warning: test environment \"$env_name\" doesn't exist, nothing to destroy."
+		return 1
```

Comment:
I'm not as familiar with bats as you, but I think bats will set the test as fail if you return 1 from any function called by a test, so this is different from what is commented.
It looks like this isn't a problem for teardown, as you said. So just change the comment (if this is really the case)

Reply:
I'm not sure I understand your comment @otaviobp. Returning 1 to mark the test as failed if something in the teardown fails is ok, but exiting the script from the teardown is not since it will live the test with no result.

Reply:
ok. got that :)